### PR TITLE
Migrate the use of fsnotify to fswatcher in cert_watcher.go

### PR DESCRIPTION
**Pull request by haanhvu**
Signed-off-by: haanhvu <haanh6594@gmail.com>

## Which problem is this PR solving?
First step for #3924 as discussed [here](https://github.com/jaegertracing/jaeger/issues/3924#issuecomment-1424754016)

### CODE REVIEW

**Comment by haanhvu** [comment]
@yurishkuro You said [here](https://github.com/jaegertracing/jaeger/issues/3924#issuecomment-1424754016) that we need to make both `static_handler.go` and `cert_watcher.go` use `fswatcher`. But I already see `static_handler.go` use it. So I just make `cert_watcher.go` use it in this PR.

Please help review!

**Review by yurishkuro — APPROVED** [review]
👍

File: pkg/fswatcher/fs_watcher.go [hunk 1]
```diff
@@ -34,6 +35,11 @@ func (f *fsnotifyWatcherWrapper) Add(name string) error {
 	return f.fsnotifyWatcher.Add(name)
 }
 
+// Close closes the watcher.
+func (f *fsnotifyWatcherWrapper) Close() error {
```
Comment (yurishkuro) [inline pkg/fswatcher/fs_watcher.go:39]: please add a test for this func

Comment (haanhvu) [reply pkg/fswatcher/fs_watcher.go:39]: Ok I'll do it.

Is there any way to check Codecov failure locally? I ran `make test` and as I can remember it didn't report it.

Comment (yurishkuro) [reply pkg/fswatcher/fs_watcher.go:39]: Not to my knowledge. You can only compare package level coverage before and after the change with go test -cover.

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
🎉   Thanks!

**Pull request merged**